packages/common-all/src/arrays.ts

```diff
@@ -0,0 +1,14 @@
+export {};
+
+/** Array extension methods. */
+declare global {
+  interface Array<T> {
+    /** Extension method returns true when array has elements, false otherwise. */
+    hasElements(): boolean;
+  }
+}
+
+Array.prototype.hasElements = function <T>(): boolean {
+  const arr = this as Array<T>;
+  return arr.length > 0;
+};
```

Comment:
I'm not sure if we should be extending native objects for array empty checks. 

We are already using lodash extensively, which provides `_.isEmpty` that could be used instead of extending the native array.

In any case, I think a simple `if (someArray.length)` should suffice.
`if (Array.isArray(someArray) && someArray.length)` if you want to be extra careful.

Reply:
Sure, for now I can just use `_.isEmpty(arrayValue)` although [imho]: `arrayValue.hasElements()` reads nicer.

On using length in an if statement as above `if (someArray.length)` I would disagree with such style since it is a bit on the implicit side meaning takes advantage of typescripts/javascripts subtlety about truthiness evaluation of 0 being false. And would rather write `if (someArray.length === 0)` to be more explicit.

Reply:
Lets avoid monkey patching javascript built-ins. This makes code harder to follow since its no longer standard javascript. 

As for negations - generally agree that negative conditionals should be avoided when possible but a more important principle we follow is to reduce complexity in the code base. ideally, there should be one way of doing things. so rather have `!_.isEmpty` be the way for checking array fullness instead of adding a new method

Reply:
Since we also have some usages of arr.length (Eg. https://github.com/dendronhq/dendron/blob/master/packages/plugin-core/src/external/memo/utils/utils.ts#L206) and it avoids a double negative using that for now.